Wire-Protocol.rst

```diff
@@ -1,49 +1,102 @@
+``sender_info.package``  string  All          Software package used to send the message
+``sender_info.exe``      string  All          Full path of the executable used to send the message
+``sender_info.version``  string  All          Sender package version
+``sender_info.commit``   string  All          Git commit of the sender package
+``sender_info.hostname`` string  All          Name of the host computer that sends the message
+``sender_info.username`` string  All          User responsible for sending the message
+``retcode``              integer Replies      Machine-interpretable status; see :ref:`retcodes`
+``return_msg``           string  Replies      Human-readable explanation of the return code
+======================== ======= ============ ===========================================
 
 
 Payload
 -------
 
-The payload is the contents of the message, and it comprises the AMQP message body.  It is optional in all types of dripline messages.
+The payload is the content of the message, and it comprises the AMQP message body.  It is optional in all types of dripline messages.
 
 The payload should be formatted in `JSON <http://json.org>`_.  The encoding (``application/json``) should be specified in the ``content_encoding`` field of the AMQP message.
 
 
+.. _splitmsg:
+
+Split Messages
+==============
+
+A dripline message can be split across multiple AMQP messages to accommodate large payloads.  
+The AMQP messages that comprise a single dripline message are called "chunks."  
+The headers for all chunks comprising a single dripline message are identical, with the exception of ``message-id``.  
+The payload, encoded as a JSON string, is split into substrings based on a maximum size, and each substring is assigned as the payload for a single chunk in order of position in the original string.
+
+The ``message-id`` header field is used to properly reconstruct the full message.  
+The structure of the ``message-id`` for a split message is ``UUID/[chunk number]/[total chunks]``.  
+The ``UUID`` is a single unique identifier for the dripline message, and is duplicated in each chunk.  
+The ``chunk number`` describes the position of the chunk in the full message.  
+The `total chunks`` is the number of chunks into which the message was divided.
+
+For unsplit messages the ``message-id`` field may either be ``UUID`` or ``UUID/0/1``.
+
+
+.. _message_types:
+
+Message Types
+=============
+
+There are three types of dripline messages:
+
+:Request: messages sent from one node to another in a dripline mesh to perform a specific action.
+:Reply: messages sent in reply to a request.
+:Alert: messages broadcast from a node without regard to whether other nodes are listening.
+
+
+.. _message_op:
+
+Message Operations
+==================
+
+Request messages have four possible operations:
+
+:Set: set a value
+:Get: get a value
+:Run: commence running
+:Command: perform a command
+
```

Comment:
are we deprecating send (https://github.com/driplineorg/dripline-cpp/blob/02442ef23c41e453279d730f2a05e42a091c8d0d/library/dripline_constants.hh#L34)?

I think existing cases where that is used in dripline 2 could just be commands.

Reply:
I'm fine with keeping it in if there's a purpose.  Currently the C++ doesn't handle it.  It's pretty trivial to add another op-handler function.  Your call, since you're more familiar with the use cases.

Reply:
I'm happy to remove it, I now realize that it is a bit weird to have it as a special case (similar to our removal of `config`).

I should ask a similar question about `run`... do we actually use `run` in psyllid or do we just send a `cmd` which is `start-run`?

Reply:
In Psyllid a `run` request and a `start-run` `cmd` are handled in exactly the same way.

Are there any other use cases for that from the python side?

Reply:
I don't think there are any cases in python; if there are a `run` will be equivalent to a particular specifier on a `cmd`

Reply:
I'll remove `run` and `send` from the protocol.